core/src/utils/native/capacitor.ts

```diff
@@ -0,0 +1,13 @@
+import type { CapacitorInstance } from '@capacitor/core/types/definitions-internal';
```

Comment:
Is it safe to rely on internal definitions?

Reply:
Ideally we could use `CapacitorGlobal` (the public type definition that Capacitor exports). There is two problems that exist with it and our implementation however:

1. We make use of "private" APIs: https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/native/status-bar.ts#L24
2. `window.Capacitor.Plugins` is considered deprecated on the public interface, but is treated as stable on the internal type definition. It doesn't seem that Capacitor has intentions of removing that property in the future.

I think there is risk in both directions:
- Using the internal type does mean they could move or change the type. This would result in a build error when updating the dependency. It should not result in shipped regressions. 
- Using the public type, we would need to disable type checking for the areas of the implementation that specifically use private APIs (just status bar right now). This puts us into a similar scenario as exists on main, where that API could be removed or changed and we would not know about it until someone reported an issue.

Reply:
If we used `CapacitorGlobal` we could use `CapacitorGlobal.isPluginAvailable` it sounds like: https://github.com/ionic-team/capacitor/blob/06fc5f1d819ffaffb57423a3f459449b0ec08c0f/core/src/definitions.ts#L31C3-L31C20

This would let us use public definitions for all but [the status bar](https://github.com/ionic-team/ionic-framework/blob/096d9cc931dfe3a1b4b39d6652112ef209be7147/core/src/utils/native/status-bar.ts#L24). As for that line, we can continue to use `win.Capacitor.PluginHeaders` and then make a tech debt ticket to officially drop support for older Capacitor versions in Ionic 8 or something similar.

Reply:
Sounds like a good compromise 👍 I'll create a ticket to track. We will effectively be dropping support for the status bar appearance automatically updating back to the default style when using the card modal on Capacitor 2 with Ionic v8.